### ISSUE TITLE
Spell out that `bower` also needs to be installed 

### ISSUE THREAD

**Issue by zzr994** [balinterdi/rarwe-issues#150]
I was following along in chapter 2 to install node, ember etc.

1. $ npm install -g ember-cli
2. $ npm install -g bower
3. $ ember new rarwe

Even though the second command is not in the book, I had to use it without it bower could not be found.  Has others faced the same issue?

See sequence of steps below:
C:\EmberApps>ember -v
version: 1.13.14
node: 0.12.9
npm: 2.14.10
os: win32 x64

C:\EmberApps>bower install
'bower' is not recognized as an internal or external command,
operable program or batch file.

C:\EmberApps>npm install -g bower
C:\Users\lester\AppData\Roaming\npm\bower -> C:\Users\lester\AppData\Roaming\npm
\node_modules\bower\bin\bower
bower@1.7.2 C:\Users\lester\AppData\Roaming\npm\node_modules\bower
+-- semver-utils@1.1.1

C:\EmberApps>ember -v
version: 1.13.14
node: 0.12.9
npm: 2.14.10
os: win32 x64

C:\EmberApps>ember new rarwe
version: 1.13.14
installing app
  create .bowerrc
  create .editorconfig
  create .ember-cli
  create .jshintrc
  create .travis.yml
  create .watchmanconfig
  create README.md
  create app\app.js
  create app\components\.gitkeep
  create app\controllers\.gitkeep
  create app\helpers\.gitkeep
  create app\index.html
  create app\models\.gitkeep
  create app\router.js
  create app\routes\.gitkeep
  create app\styles\app.css
  create app\templates\application.hbs
  create app\templates\components\.gitkeep
  create bower.json
  create config\environment.js
  create ember-cli-build.js
  create .gitignore
  create package.json
  create public\crossdomain.xml
  create public\robots.txt
  create testem.json
  create tests\.jshintrc
  create tests\helpers\destroy-app.js
  create tests\helpers\module-for-acceptance.js
  create tests\helpers\resolver.js
  create tests\helpers\start-app.js
  create tests\index.html
  create tests\integration\.gitkeep
  create tests\test-helper.js
  create tests\unit\.gitkeep
  create vendor\.gitkeep
Successfully initialized git.
Installed packages for tooling via npm.
Failed to execute "git ls-remote --tags --heads git://github.com/ember-cli/loade
r.js.git", exit code of #128
fatal: unable to connect to github.com:
github.com[0: 192.30.252.131]: errno=No error

Error: Failed to execute "git ls-remote --tags --heads git://github.com/ember-cl
i/loader.js.git", exit code of #128
fatal: unable to connect to github.com:
github.com[0: 192.30.252.131]: errno=No error

    at createError (C:\Users\lester\AppData\Roaming\npm\node_modules\ember-cli\n
ode_modules\bower\lib\util\createError.js:4:15)
    at ChildProcess.<anonymous> (C:\Users\lester\AppData\Roaming\npm\node_module
s\ember-cli\node_modules\bower\lib\util\cmd.js:102:21)
    at ChildProcess.emit (events.js:110:17)
    at maybeClose (child_process.js:1016:16)
    at Process.ChildProcess._handle.onexit (child_process.js:1088:5)

C:\EmberApps>bower install
? May bower anonymously report usage statistics to improve the tool over time? N
o
bower ENOENT        No bower.json present

C:\EmberApps>cd rarwe

C:\EmberApps\rarwe>bower install
bower not-cached    git://github.com/ember-cli/loader.js.git#3.4.0
bower resolve       git://github.com/ember-cli/loader.js.git#3.4.0
bower not-cached    git://github.com/components/ember.git#2.2.0
bower resolve       git://github.com/components/ember.git#2.2.0
bower not-cached    git://github.com/ember-cli/ember-cli-shims.git#0.0.6
bower resolve       git://github.com/ember-cli/ember-cli-shims.git#0.0.6
bower not-cached    git://github.com/ember-cli/ember-cli-test-loader.git#0.2.1
bower resolve       git://github.com/ember-cli/ember-cli-test-loader.git#0.2.1
bower not-cached    git://github.com/components/ember-data.git#2.0.0
bower resolve       git://github.com/components/ember-data.git#2.0.0
bower not-cached    git://github.com/ember-cli/ember-load-initializers.git#0.1.7

bower resolve       git://github.com/ember-cli/ember-load-initializers.git#0.1.7

bower not-cached    git://github.com/rwjblue/ember-qunit-builds.git#0.4.16
bower resolve       git://github.com/rwjblue/ember-qunit-builds.git#0.4.16
bower not-cached    git://github.com/dockyard/ember-qunit-notifications.git#0.1.
0
bower resolve       git://github.com/dockyard/ember-qunit-notifications.git#0.1.
0
bower not-cached    git://github.com/stefanpenner/ember-jj-abrams-resolver.git#~
0.1.20
bower resolve       git://github.com/stefanpenner/ember-jj-abrams-resolver.git#~
0.1.20
bower not-cached    git://github.com/jquery/jquery-dist.git#1.11.3
bower resolve       git://github.com/jquery/jquery-dist.git#1.11.3
bower not-cached    git://github.com/jquery/qunit.git#~1.20.0
bower resolve       git://github.com/jquery/qunit.git#~1.20.0
bower ECMDERR       Failed to execute "git ls-remote --tags --heads git://github
.com/ember-cli/ember-cli-test-loader.git", exit code of #128 fatal: unable to co
nnect to github.com: github.com[0: 192.30.252.128]: errno=No error

Additional error details:
fatal: unable to connect to github.com:
github.com[0: 192.30.252.128]: errno=No error

C:\EmberApps\rarwe>

**Comment by zzr994** [comment]
While creating the new App, it complained about GIT not being found. Hence I had to install GIT first.

**Comment by zzr994** [comment]
Why am I getting the following error?

Error: Failed to execute "git ls-remote --tags --heads git://github.com/ember-cl
i/loader.js.git", exit code of #128
fatal: unable to connect to github.com:
github.com[0: 192.30.252.131]: errno=No error

**Comment by fknoll** [comment]
@zzr994 I am also working with Ember.js on a Windows PC (Windows 7, 64 bit). I am up running using "Node.js Command Prompt": 
```C:\Windows\System32\cmd.exe /k "C:\Program Files\nodejs\nodevars.bat"```

When starting out with Ember.js some months ago, I also had problems with Bower, but unfortunately, I do not remember the details.

**Comment by balinterdi** [comment]
@zzr994 From the error it seems like Github was down at the time you tried, do you get this consistently? Unfortunately, I can't really help you here as I don't have access to a Windows machine.

**Comment by zzr994** [comment]
Thanks Balint,

I found and fixed the issue on my windows machine.
The fix was taken from the reference below:
http://stackoverflow.com/questions/15669091/bower-install-using-only-https

The error message that I was getting is as follows. 
It shows the failure of Bower to install components from the github repository.
This is due to the firewall on the windows machine.

Error: Failed to execute "git ls-remote --tags --heads git://github.com/componen
ts/ember-data.git", exit code of #128
fatal: unable to connect to github.com:
github.com[0: 192.30.252.131]: errno=No error

If you notice the bower is using the git protocol, git://github.com/....
If you try the same URL through the browser using https (https:/githumb.com/....).
It works.  So the solution would be to force git to use https instead of git protocol.
This can be done using the command below from the windows command prompt.
Cut and paste the command below exactly as it appears below:

git config --global url."https://".insteadOf git://

This however did not work on my windows. This is because of the following issue.

--- cut and pasted from the stackoverflow reference ---
"On Windows machine, the global configuration file is .gitconfig under user's home folder, 
e.g. C:\Users[username]. However if %HOME% is not defined, git will using %HOMEDRIVE% 
while git from bower will use %USERPROFILE% instead. Whereas these two variables might 
be different. On my machine, one is U:, the other is C:\Users\myusername. 
So the bower still used git:// whatever I tried. It took me a while to figure this out. 
So write it down in case anybody falls into the same situation. – AnthonyY Dec 16 '14 at 21:59"  

So I went and manually added the following to the end of the file: c:\Users\username\.gitconfig
[url "https://"]	
	insteadOf = git://

Now from the Node command prompt the following command will work.
ember new rarwe

I am putting the details here in the hope that it will help others who run into the same issue.
